core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1197,6 +1196,10 @@ public ServerBuilder tlsProvider(TlsProvider tlsProvider) {
         requireNonNull(tlsProvider, "tlsProvider");
         this.tlsProvider = tlsProvider;
         tlsConfig = null;
+
+        if (tlsProvider.autoClose()) {
```

Comment:
Note) I understood if a `TlsProvider` is overwritten, it will still be closed.

e.g.
```java
sb.https(0)
  .tlsProvider(tlsProvider)
  // overwrite
  .tlsProvider(tlsProvider2);
// both tlsProvider, tlsProvider2 are closed
```